pkg/cloud/services/compute/referenced_resources.go

```diff
@@ -0,0 +1,39 @@
+/*
```

Comment:
I wish we could put it out of compute package so it's not tight to compute.
Also unit tests.

Reply:
Test added, found a bug.

Where would you propose to put this module? I took a look, but nowhere it really fits better than here, as it's MachineSpec passed there.